src/cryptoadvance/specter/liquid/wallet.py

```diff
@@ -502,6 +503,49 @@ def adjust_fee(self, psbt, fee_rate):
             * (tx_full_size / psbt["tx"]["vsize"])
         )
 
+    def addresses_info(self, is_change):
+        """Create a list of (receive or change) addresses from cache and retrieve the
+        related UTXO and amount.
+        Parameters: is_change: if true, return the change addresses else the receive ones.
+        """
+
+        addresses_info = []
+
+        addresses_cache = [
+            v for _, v in self._addresses.items() if v.change == is_change
+        ]
+
+        for addr in addresses_cache:
+
+            addr_utxo = 0
+            addr_amount = 0
+            addr_assets = []
+
+            for utxo in [
+                utxo
+                for utxo in self.full_utxo
+                if to_unconfidential(utxo["address"]) == to_unconfidential(addr.address)
+            ]:
+                addr_amount = addr_amount + utxo["amount"]
+                addr_utxo = addr_utxo + 1
+                if utxo.get("asset") not in addr_assets:
+                    addr_assets.append(utxo.get("asset"))
+
+            addresses_info.append(
+                {
+                    "index": addr.index,
+                    "address": addr.address,
+                    "label": addr.label,
+                    "amount": addr_amount,
+                    "used": bool(addr.used),
+                    "utxo": addr_utxo,
+                    "type": "change" if is_change else "receive",
+                    "assets": addr_assets,
```

Comment:
So the amount here in case of multiple assets is the sum of all assets?
This means there is no way to learn about the balances of different assets from this endpoint.
Maybe it would be better to change `assets` from a list to a dict with `{asset:amount}`?
Then we could potentially show more detailed info about the balance on the address in the future, it also could be used by other apps in the API calls.

Reply:
Ok changed to this